### PR TITLE
Fix support for torch Module type hints in LightningCLI

### PR DESCRIPTION
**Pull request by mauvilsa**
## What does this PR do?

Fixes a problem mentioned in https://github.com/PyTorchLightning/pytorch-lightning/issues/7774#issuecomment-851813825. It does not resolve the whole issue, only that single comment.

## Before submitting
- [X] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [X] Did you make sure to update the documentation with your changes? (if necessary)
- [X] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [X] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by mauvilsa** [comment]
I did verify that the tests were successful locally. Not sure why it is failing.

**Comment by carmocca** [comment]
I can reproduce the failure locally

**Comment by mauvilsa** [comment]
I guess torchvision has changed a log between versions. In https://github.com/PyTorchLightning/pytorch-lightning/pull/7807/checks?check_run_id=2728907218 the error is `"torchvision.transforms.Resize" is not a subclass of Module`. I upgraded torchvision in my venv and now it does fail but the error is different `('cannot represent an object', <InterpolationMode.BILINEAR: 'bilinear'>)`. I don't know with which version it was successful for me.

I will look into this last error and probably also change the test to use something a bit more stable.

File: CHANGELOG.md [hunk 1]
```diff
@@ -182,6 +182,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed a bug where `precision=64` with `accelerator='ddp_spawn'` would throw a pickle error ([#6924](https://github.com/PyTorchLightning/pytorch-lightning/pull/6924))
 
 
+- Fixed support for torch Module type hints in `LightningCLI` ([#7807](https://github.com/PyTorchLightning/pytorch-lightning/pull/7807))
```
Comment (carmocca) [inline CHANGELOG.md:185]:
```suggestion
- Fixed support for `torch.nn.Module` type hints in `LightningCLI` ([#7807](https://github.com/PyTorchLightning/pytorch-lightning/pull/7807))
```

**Review by carmocca — APPROVED** [review]

File: requirements/extra.txt [hunk 2]
```diff
@@ -7,4 +7,4 @@ torchtext>=0.5
 # onnx>=1.7.0
 onnxruntime>=1.3.0
 hydra-core>=1.0
-jsonargparse[signatures]>=3.12.0
+jsonargparse[signatures]>=3.13.1
```
Comment (awaelchli) [inline requirements/extra.txt:10]: is this the fix?

**Review by awaelchli — APPROVED** [review]

File: requirements/extra.txt (hunk 2 above)
Comment (mauvilsa) [reply requirements/extra.txt:10]: Yes, the fix is there.

**Review by justusschock — APPROVED** [review]
